components/canvas/canvas_paint_thread.rs

```diff
@@ -795,6 +790,11 @@ fn is_zero_size_gradient(pattern: &Pattern) -> bool {
     false
 }
 
+fn premultiply_channel_by_alpha(channel: u8, alpha: u16) -> u8 {
```

Comment:
I felt like this could be spun out into a helper, not least because those `((((`s looked potentially confusing, but happy to put it back in `put_image_data`.